src/main/kotlin/org/rust/ide/formatter/RustLanguageCodeStyleSettingsProvider.kt

```diff
@@ -0,0 +1,57 @@
+package org.rust.ide.formatter
+
+import com.intellij.application.options.IndentOptionsEditor
+import com.intellij.application.options.SmartIndentOptionsEditor
+import com.intellij.lang.Language
+import com.intellij.psi.codeStyle.CodeStyleSettingsCustomizable
+import com.intellij.psi.codeStyle.CommonCodeStyleSettings
+import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider
+import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider.SettingsType.*
+import org.rust.ide.utils.loadCodeSampleResource
+import org.rust.lang.RustLanguage
+
+class RustLanguageCodeStyleSettingsProvider : LanguageCodeStyleSettingsProvider() {
+    override fun getLanguage(): Language = RustLanguage
+
+    override fun getCodeSample(settingsType: SettingsType): String =
+        // TODO Provide more suitable code samples for specific settings types
+        when (settingsType) {
+            BLANK_LINES_SETTINGS,
+            SPACING_SETTINGS,
+            WRAPPING_AND_BRACES_SETTINGS,
+            INDENT_SETTINGS -> CODE_SAMPLE
+
+            else            -> ""
+        }
+
+    override fun customizeSettings(consumer: CodeStyleSettingsCustomizable, settingsType: SettingsType) {
+        when (settingsType) {
+            BLANK_LINES_SETTINGS         -> {
+            }
+
+            SPACING_SETTINGS             -> {
+            }
+
+            WRAPPING_AND_BRACES_SETTINGS -> {
+                consumer.showStandardOptions("RIGHT_MARGIN");
+            }
+
+            INDENT_SETTINGS              -> {
+            }
+
+            else                         -> {
+            }
```

Comment:
This piece is a relict from the times when I planned to implement whole formatter at once. I can leave todos or remove this if you wish to.